server/src/embedded-languages/bash-support.ts

```diff
@@ -25,10 +25,26 @@ export const generateBashEmbeddedLanguageDoc = (analyzedDocument: AnalyzedDocume
         return false
     }
   })
-  insertTextIntoEmbeddedLanguageDoc(embeddedLanguageDoc, 0, 0, shebang)
+  insertShebang(embeddedLanguageDoc)
+  if (pokyFolder !== undefined) {
+    insertBashTools(embeddedLanguageDoc, pokyFolder)
+  }
   return embeddedLanguageDoc
 }
 
+const insertShebang = (embeddedLanguageDoc: EmbeddedLanguageDoc): void => {
+  insertTextIntoEmbeddedLanguageDoc(embeddedLanguageDoc, 0, 0, shebang)
+}
+
+const insertBashTools = (embeddedLanguageDoc: EmbeddedLanguageDoc, pokyFolder: string): void => {
+  const bashTools = [
+    `. ${pokyFolder}/meta/classes-global/logging.bbclass`,
+    `. ${pokyFolder}/meta/classes-global/base.bbclass`,
```

Comment:
base.bbclass is not a valid file for bash. I wonder how bash-ide and shellcheck manage to get something from it. logging.bbclass is a bit closer but also has problems. I think we should extract the bash regions from them only. Either through tree-sitter but it can be tricky? Otherwise I think it's acceptable to have our own logging.sh and base.sh copies in our sources and source them.

Reply:
Yeah, I am also surprised it works, but I've not seen any problem with the current solution, so I guess it is fine?

If we ship our own logging.sh and base.sh, we won't be able to map the positions of these files to the original document. We'll have to disable the "jump to definition" to these specific files, and edit path showed in "hover".  Maybe we're better work on a more general solution that would handle "include", "inherit" and "require"?

Reply:
It's better to have the original positions as well. It looks like shellcheck and bash-ide can properly handle all the non bash lines without question so we can keep this as is. We should be mindful that this could break at some point, which wouldn't be critical for users and quickly detected by the CI. What I suggest would be to add a comment in the code suggesting to switch to our own logging.sh/base.sh and disable definitions if this ever breaks.

Yes you could also add a general solution for include statements and see if it tends to work or break. The clean way would be to extract the bash regions from them, but could be slow and create many more documents. We also currently don't have the accurate list of includes untill Ziwei's recipe scan PR is merged. Without it we would have most results, but some like base.bbclass require Ziwei's scan to be detected since they use other external include mecanisms.